src/components/main/Main.tsx

```diff
@@ -36,17 +48,17 @@ const Main: FC = () => {
   }
 
   useEffect(() => {
-    if (position.longitude) {
-      fetchData(position.longitude, position.latitude);
+    if (resultLongitude && resultLongitude !== '0') {
+      fetchData(resultLongitude, resultLatitude);
     }
-  }, [position.longitude]);
+  }, [resultLongitude, resultLatitude]);
 
   useEffect(() => {
     if (selectDays) {
       getWeather<{ forecast: DailyWeatherType[] }>(
         '/forecast/daily/',
-        position.longitude,
-        position.latitude,
+        resultLongitude,
+        resultLatitude,
         { periods: selectDays, dataset: 'full' }
       ).then(res => setDailyWeather(res.forecast));
     }
```

Comment:
Do you mean useEffect with selectDays?
In this case I do request only when change selected days.

Reply:
resultLongitude and resultLatitude should be mentioned in  second useEffect deps array

Reply:
and actually, you can move this request to function  like `handleDaysSelected`, that will be placed instead of `onChange={e => setSelectDays(e.target.value)}`, because it's the only way to update `selectDays` state